src/Sylius/Component/Core/OrderProcessing/OrderShipmentProcessor.php

```diff
@@ -69,6 +69,12 @@ public function process(OrderInterface $order)
      */
     private function getOrderShipment(OrderInterface $order)
     {
+        if ($order->isEmpty()) {
+            $order->removeShipments();
```

Comment:
We shouldn't do it in `getOrderShipment` method but rather in the beginning of a `process`method.